pgfinder/matching.py

```diff
@@ -174,49 +168,28 @@ def modification_generator(filtered_theo_df: pd.DataFrame, mod_type: str) -> pd.
         )
     return obs_theo_muropeptides_df
 
-
 def matching(ftrs_df: pd.DataFrame, matching_df: pd.DataFrame, set_ppm: int):
-    """Match theoretical masses to observed masses within ppm tolerance.
-
-    Parameters
-    ----------
-    ftrs_df: pd.DataFrame
-        Features DataFrame
-    matching_df: pd.DataFrame
-        Matching DataFrame
-    set_ppm: int
-
-    Returns
-    -------
-    pd.DataFrame
-        Dataframe of matches.
-    """
-    raw_data = ftrs_df.copy()
-    # Data validation
-    if ("Monoisotopicmass" not in matching_df.columns) | ("Structure" not in matching_df.columns):
-        print(
-            'Header of csv files must have column named "Monoisotopic mass" and another column named "Structure"!!!  Make note of capitalized letters and spacing!!!!'
-        )
-
-    # Generates dataframe with matched structures
-    for x, row in raw_data.iterrows():
-        # Observed monoisotopic mass
-        mw = row.mwMonoisotopic
-        # ppm tolerance value
-        t_tol = calc_ppm_tolerance(mw, set_ppm)
-        # create dataframe with values from matching_df within tolerance to observed monoisotopic mass
-        t_df = matching_df[
-            (matching_df["Monoisotopicmass"] >= mw - t_tol) & (matching_df["Monoisotopicmass"] <= mw + t_tol)
-        ]
-        # Populate inferred structure and theo_mwMonoisotopic columns with matched values
-        if not t_df.empty:
-            raw_data.loc[x, "inferredStructure"] = ",".join(t_df.Structure.values)
-            raw_data.loc[x, "theo_mwMonoisotopic"] = ",".join(
-                [f"{x:0.4f}".rstrip("0") for x in t_df.Monoisotopicmass.values]
-            )
-
-    return raw_data
-
+    molecular_weights = matching_df[["Structure", "Monoisotopicmass"]]
+    matches_df = pd.DataFrame()
+
+    for s, m in molecular_weights.itertuples(index=False):
+        # FIXME: I'm not sure if it's better to convert everything to float or
+        # to convert everthing to Decimal instead
+        m = float(m)
```

Comment:
@bobturneruk The reason that forcing is there is indeed because the tests fail without it! I think when either multimers or modified residues are plugged back into the program after the initial monomer search, they have masses in `Decimal`s now!

The `float()` is a No-op for monomers, but converts from `Decimal` when the same function is called with a generated mass-list I think :)

Reply:
I might keep the float() force in for now since I need it to pass the tests, but definitely worth looking at going forward!

Reply:
Fair enough - it works as is. A problem for another day.